pkg/printers/internalversion/printers.go

```diff
@@ -574,24 +574,34 @@ func printPod(pod *api.Pod, options printers.PrintOptions) ([]metav1beta1.TableR
 	}
 	if !initializing {
 		restarts = 0
+		hasCompleted := false
+		hasRunning := false
 		for i := len(pod.Status.ContainerStatuses) - 1; i >= 0; i-- {
 			container := pod.Status.ContainerStatuses[i]
 
 			restarts += int(container.RestartCount)
 			if container.State.Waiting != nil && container.State.Waiting.Reason != "" {
 				reason = container.State.Waiting.Reason
 			} else if container.State.Terminated != nil && container.State.Terminated.Reason != "" {
-				reason = container.State.Terminated.Reason
+				if container.State.Terminated.Reason == "Completed" {
+					hasCompleted = true
+				} else {
+					reason = container.State.Terminated.Reason
+				}
 			} else if container.State.Terminated != nil && container.State.Terminated.Reason == "" {
 				if container.State.Terminated.Signal != 0 {
 					reason = fmt.Sprintf("Signal:%d", container.State.Terminated.Signal)
 				} else {
 					reason = fmt.Sprintf("ExitCode:%d", container.State.Terminated.ExitCode)
 				}
 			} else if container.Ready && container.State.Running != nil {
+				hasRunning = true
 				readyContainers++
 			}
 		}
+		if reason == string(api.PodRunning) && hasCompleted && !hasRunning {
```

Comment:
What if the pod's status is `Completed`? 
This line would make it so that Completed pods would appear with a `Succeeded` status, rather than `Completed`.

Why not simplify to this (or something similar)?

```go
...
if !initializing {
    restarts = 0
    hasRunning := false
    for i := ... {
        ...
        // last "else if" in loop 
        } else if container.Ready && container.State.Running != nil {
            hasRunning = true
            readyContainers++
        }
    }

    // revert status back to "Running" if there is at least onr
    // container still reporting the "Running" status
    if reason == "Completed" && hasRunning {
        reason = "Running"
    }
}
```

Reply:
Thank you. 

Sounds great.
Change the pod status codes according to your comment.